tutorials/single-page-tutorials/nested-nfts-with-wallet-lib.md

```diff
@@ -0,0 +1,423 @@
+
+![Warrior avatar](./images/warrior-avatar.png)
+![Warrior mask](./images/warrior-mask.png)
+![Warrior weapon](./images/warrior-weapon.png)
+
+---
+
+## Scripts
+
+### Create NFTs
+
+Create a new file `create-nfts.js` and add the following code.
+
+:::note
+
+We broke the code into separate snippets to help you understand it better. To make it work, copy all code snippets one after another into the file that you have just created.
+
+:::
+
+#### 1. Imports and parameters
+
+```javascript
+// Libraries
+const { utf8ToHex } = require('@iota/client');
+const { AccountManager } = require('@iota/wallet');
+const IPFS = require('ipfs-core');
+const fs = require('fs');
+
+// Environment variables
+require('dotenv').config({ path: './.env' });
+const password = process.env.SH_PASSWORD;
+const accountName = process.env.ACCOUNT_NAME;
+
+// Network configuration
+const { networkConfig } = require('./networkConfig.js');
+const explorerURL = networkConfig.explorer;
+
+// For better readability, some console output will be printed in a different color
+const consoleColor = '\x1b[36m%s\x1b[0m';
+```
+
+#### 2. Start local IPFS node
+
+The function `startIpfsNode()` starts a local IPFS node for file upload.
+
+```javascript
+// This function starts a local IPFS node for file upload
+async function startIpfsNode() {
+  console.log('\n');
+  console.log(consoleColor, `Start local IPFS node for upload:`);
+
+  let node;
+  if (!node) {
+    node = await IPFS.create({
+      repo: `ipfs_node`,
+    });
+  }
+  return node;
+}
+```
+
+#### 3. Upload file to IPFS
+
+The function `uploadToIPFS()` receives a running ipfs node instance and uploads a file to IPFS from a given file path.
+
+```javascript
+// This function receives a running ipfs node instance and a file path for the upload
+async function uploadToIPFS(ipfsNode, filePath) {
+  try {
+    // Read file from path
+    const file = fs.readFileSync(filePath);
+
+    // Upload file to IPFS
+    const fileAdded = await ipfsNode.add(file);
+    const contentIdentifier = fileAdded.path;
+
+    console.log('\n');
+    console.log(consoleColor, `Your file was successfully uploaded to IPFS:`);
+    console.log(`https://ipfs.io/ipfs/${contentIdentifier}`, '\n');
+
+    return contentIdentifier;
+  } catch (error) {
+    console.error('IPFS upload error', error);
+  }
+}
+```
+
+#### 4. Mint NFT
+
+The function `mintNFT()` takes a NFT name, the previously created and unique IPFS identifier and an unlocked Stronghold account in order to mint an NFT on the configurated network. The metadata will be stored in the immutable metadata field, meaning it will remain unchanged for as long as the NFT itself exists on the network. Also note that the function does not pass a specific address to the Stronghold Account method `mintNfts()`. By default the method will mint all NFTs on the first address of the used Stronghold account, as long as it holds enough funds for Storage Deposit.
```

Comment:
Referring to previous comments. In case we choose to not build on the wallet tutorial but make this tutorial more stand-alone, we actually do want to pass the address that the reader funded, one way or another.